sharktuner/sharktuner/dispatch_constraints.py

```diff
@@ -502,6 +514,25 @@ def set_cdim_tile_sizes(tile_sizes, contraction_dims, csizes):
             [lookup(v) for v in k_vars],
         )
 
+        required_padding = any(
+            p[-1] % i != 0
+            for p, i in zip(problem_size.MNK, intrinsic_mnk_shape, strict=True)
+        )
+        promote_operands = [0, 1]
+        if required_padding:
+            # TODO: Remove promotion of operand 2 once codegen supports handling padded outputs without promotion.
+            promote_operands = [0, 1, 2]
+            workgroup_tile_m, workgroup_tile_n, _ = workgroup_tile_sizes
+            _, _, reduction_tile_k = reduction_tile_sizes
+            _, _, mma_intrinsic_k = mma_attr.mnk_shape
+            padding = [
+                workgroup_tile_m,
+                workgroup_tile_n,
+                reduction_tile_k * mma_intrinsic_k,
+            ]
+        else:
+            padding = None
```

Comment:
nit: It is a bit more clear and concise if `padding` is defined outside the if.
```suggestion
        padding = None
        if required_padding:
            # TODO: Remove promotion of operand 2 once codegen supports handling padded outputs without promotion.
            promote_operands = [0, 1, 2]
            workgroup_tile_m, workgroup_tile_n, _ = workgroup_tile_sizes
            _, _, reduction_tile_k = reduction_tile_sizes
            _, _, mma_intrinsic_k = mma_attr.mnk_shape
            padding = [
                workgroup_tile_m,
                workgroup_tile_n,
                reduction_tile_k * mma_intrinsic_k,
            ]
```